src/build123d/topology.py

```diff
@@ -2625,47 +2999,77 @@ def _bool_op(
         Returns:
 
         """
+        # Find the highest order class from all the inputs Solid > Vertex
+        order_dict = {type(s): type(s).order for s in [self] + list(args) + list(tools)}
```

Comment:
You're going over `args` and `tools` again below so you don't want to just consume them here in case they're lazy
```suggestion
        args, args2 = tee(args)
        tools, tools2 = tee(tools)
        order_dict = {type(s): type(s).order for s in (self, *args2, *tools2)}
```

Reply:
I just used lists here as the number of objects are going to be small.